### PR TITLE
Add a filter for activity contact record_type_id to the Activity Summary Report

### PR DESCRIPTION
**Pull request by alifrumin**
…ummary report

Overview
----------------------------------------
This PR adds a filter to the Activity Summary Report template so that one can choose which Activity Contact Record Type (assignee, with or source) to report on.

Before
----------------------------------------
Before this PR the Contact Name was always the Target (also referred to as the with contact).

After
----------------------------------------
Now one can choose which Activity Contact to run the report for (assignee, with or source). It defaults to Target, so unless the user changes the filter the report should work as before. 

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Report/Form/ActivitySummary.php [hunk 1]
```diff
@@ -305,6 +313,12 @@ public function from($durationMode = FALSE) {
     $assigneeID = CRM_Utils_Array::key('Activity Assignees', $activityContacts);
     $targetID = CRM_Utils_Array::key('Activity Targets', $activityContacts);
     $sourceID = CRM_Utils_Array::key('Activity Source', $activityContacts);
+    $activityContactTables = array(
+      $assigneeID => 'assignment_activity',
+      $targetID => 'target_activity',
+      $sourceID => 'source_activity',
+    );
+    $activityRole = $activityContactTables[$this->_params['activity_role_value']];
```
Comment (agh1) [inline CRM/Report/Form/ActivitySummary.php]: Maybe replace this with the following:

```php
$activityContactJoins = '';
foreach ($activityContactTables as $rId => $rAlias) {
  $activityContactJoins .= <<<HERESQL
                LEFT JOIN civicrm_activity_contact $ralias
                       ON {$this->_aliases['civicrm_activity']}.id = $ralias.activity_id AND
                          $rAlias.record_type_id = $rId
HERESQL;
}
$activityContactJoins .= <<<HERESQL
                LEFT JOIN civicrm_contact contact_civireport
                       ON {$activityContactTables[$this->_params['activity_role_value']]}.contact_id = contact_civireport.id
HERESQL;
```

and then replace the first four left joins with `{$activityContactJoins}`

Comment (agh1) [reply CRM/Report/Form/ActivitySummary.php]: Forget it--it ends up being equally long!

**Comment by eileenmcnaughton** [comment]
I'm not that comfortable with adding this without extending the testing on the report to cover it  (api_v3_ReportTemplateTest)

**Comment by alifrumin** [comment]
Thanks Eileen, I will work on a test when I get a chance

**Comment by alifrumin** [comment]
While testing this Activity Summary Report change I stumbled upon a bug on the Activity Summary Report where the count field was not totalling as expected. At first I thought the issue was with this change, but I was able to replicate the bug without this pr. I am going to close this pr and focus on understanding what's going on with that count field first.

**Comment by eileenmcnaughton** [comment]
@alifrumin ok there is a test for that report in ReportTemplateTest - big picture on Activity report IMHO is that prior to 4.4 there was no activity contact table to the temp tables & weird config exist basically to 'construct' one as the logic was never really updated

Ideally in construct we would define each contact as it's own table in the construct array (ditto related entities) & a lot of the logic would be able to go because most of the weird logic is just compensating for the way the tables are munged together in the construct array so we have to build a table to match the array (rather than the array reflecting our data structure)

**Pull request closed without merging**